README.md

```diff
@@ -214,6 +214,32 @@ In more complex cases, when `Signal`-trapping-based shutdown is not appropriate,
 
 `shutdown` is an asynchronous operation. Errors will be logged and bubbled up to the provided completion handler.
 
+### Stateful handlers
+
+In some cases it is useful to have the Start handlers return a state that can be passed on to the Shutdown handlers for shutdown.
+For example, when establishing some sort of a connection that needs to be closed at shutdown.
+
+```swift
+struct Foo {
```

Comment:
Or if it conforms to `LifecycleStartHandler` one could just pass it as?

```
 lifecycle.registerStateful(
     label: "foo",
     handler: foo
 )
```

or not possible?

Reply:
Ah i see... you only surface the new APIs really, not forcing people to impl anything. That sounds good 👍